helm/client.go

```diff
@@ -99,38 +99,18 @@ func CAFileOpt(caFile string) action.PullOpt {
 	}
 }
 
-type ConfigOpt func(*action.Configuration) error
-
-func RegistryClientConfigOpt(opts ...registry.ClientOption) ConfigOpt {
-	return func(cfg *action.Configuration) error {
-		cl, err := registry.NewClient(opts...)
-		if err != nil {
-			return fmt.Errorf("failed to create registry client: %w", err)
-		}
-
-		cfg.RegistryClient = cl
-
-		return nil
+func CertFileOpt(certFile string) action.PullOpt {
```

Comment:
IIUC this configures the client to authenticate with this client certificate (if requested by the server). Let's rename this to `ClientCertOpt` to make that clearer.
```suggestion
func ClientCertFileOpt(certFile string) action.PullOpt {
```

Reply:
Actually this is unused and cannot be configured via config so we can remove it for now.